src/io/platform/f3xx/f302x8/I2Cf302x8.cpp

```diff
@@ -0,0 +1,150 @@
+#include <EVT/io/pin.hpp>
+#include <EVT/io/platform/f3xx/f302x8/I2Cf302x8.hpp>
+
+namespace EVT::core::IO {
+
+/**
+ * Get the I2C ID associated that is associated with the given pin
+ * combination. This information is pulled from the STM32F302x8
+ * documentation with easier documentation on MBed's STM32F302r8
+ * documentation.
+ *
+ * TODO: Make modifications for this to be runnable at compile time
+ * allowing for validation of the selected pins.
+ *
+ * @param sclPin The selected I2C clock pin.
+ * @return
+ */
+static uint8_t getPortID(Pin sclPin) {
+    switch (sclPin) {
+        case Pin::PB_8:
+        case Pin::PB_6:
+        case Pin::PA_15:
+            return 1;
+        case Pin::PA_9:
+        case Pin::PF_1:
+            return 2;
+        case Pin::PA_8:
+            return 3;
+        default:
+            return 0;
+    }
+    return 0;  // Should not get here, unless bad pin given
```

Comment:
What actually happens when you give a bad pin? In the best case, nothing, but in the worst case, maybe a hard fault? You still reach ```HAL_I2C_Init()``` regardless of whether or not you enable the clocks and do everything properly.

Reply:
This will hopefully be part of a compile time check. I also plan on implementing actual error handling in the future. We should be checking the return from every call, including HAL calls for errors